src/lib/components/Skeleton/Skeleton.svelte

```diff
@@ -1,14 +1,24 @@
 <script lang="ts">
   import { onMount } from 'svelte';
-
+  import { fade } from 'svelte/transition';
   let mounted = false;
   onMount(async () => {
     mounted = true;
   });
 </script>
 
 {#if mounted}
-  <slot name="mounted" />
+  <div in:fade={{ duration: 500 }}>
+    <slot name="mounted" />
+  </div>
 {:else}
-  <slot name="loading" />
+  <div class="absolute w-full" out:fade={{ duration: 500 }}>
+    <div class="flex justify-center items-center px-4 py-2 leading-6 transition ease-in-out duration-150">
+      <svg class="animate-spin -ml-1 mr-3 h-5 w-5 text-black" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24">
+        <circle class="opacity-25" cx="12" cy="12" r="10" stroke="currentColor" stroke-width="4" />
+        <path class="opacity-75" fill="currentColor" d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z" />
+      </svg>
+      Processing ...
+    </div>
+  </div>
```

Comment:
No, this is exactly what this component should **not** do. This component does not know anything about the component it wraps. Law of demeter!

Reply:
adding to this: The placeholder does not have the same dimensions like the mounted slot. Therefore, the content gets moved down the page